### PR TITLE
[enhancement] Added RestApiException and refactored CategoryApi

### PR DESCRIPTION
**Pull request by DzmitryNev**
added RestApiException, ServiceRuntimeException, ResourceNotFoundException and appropriate handlers

also refactored CategoryApi
also added lombook dependency

### CODE REVIEW

**Comment by DzmitryNev** [comment]
1) introduction project lombook
Refactoring
2) @Controller -> @Restcontroller
3) @RequiestMapping -> @GetMapping/ @PostMapping / @PutMapping 
4) Introduce GenericRuntimeException/RestApiException/ServiceRuntimeException/ResourceNotFoundException
4) Introduce controller advices for ResourceNotFoundException(http 404) and RestApiException(http 503) - i think it is better way to handle it.
5) Introduced ErrorEntity  with errorCode and message(example of usage lombook).

File: sm-core/src/main/java/com/salemanager/shop/exception/GenericRuntimeException.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package com.salemanager.shop.exception;
+
+import lombok.Getter;
+
+public class GenericRuntimeException extends RuntimeException{
+    @Getter
+    private String errorCode;
+    @Getter
```
Comment (rahilsh) [inline sm-core/src/main/java/com/salemanager/shop/exception/GenericRuntimeException.java]: We can have a common @Getter annotation on top of class

**Comment by shopizer-ecommerce** [comment]
Hello,

Can you create a new PR on master branch, sonce only category api is done it would be preferable to start doing it for all api in master which is our upcoming release.

**Comment by DzmitryNev** [comment]
@shopizer-ecommerce
Created new PR contains only lombook changes
https://github.com/shopizer-ecommerce/shopizer/pull/285/files

This PR contains only category api changes. In the future I would like to change other apis. I think it would be better to do it step by step.

File: sm-core/src/main/java/com/salemanager/shop/exception/GenericRuntimeException.java (hunk 1 above)
Comment (DzmitryNev) [reply sm-core/src/main/java/com/salemanager/shop/exception/GenericRuntimeException.java]: Thank you. Fixed.

**Pull request merged**